docs/protocols/java_object_graph_spec.md

```diff
@@ -220,11 +293,11 @@ In most cases, all collection elements are same type and not null, elements head
 information to avoid the cost of writing it for every elements. Specifically, there are four kinds of information
 which will be encoded by elements header, each use one bit:
 
-- Whether track elements ref, use first bit `0b1` of header to flag it.
-- Whether collection has null, use second bit `0b10` of header to flag it. If ref tracking is enabled for this
+- If track elements ref, use first bit `0b1` of header to flag it.
+- If collection has null, use second bit `0b10` of header to flag it. If ref tracking is enabled for this
   element type, this flag is invalid.
-- Whether collection elements type is not declare type, use 3rd bit `0b100` of header to flag it.
-- Whether collection elements type different, use 4rd bit `0b1000` of header to flag it.
+- If collection elements type is not declare type, use 3rd bit `0b100` of header to flag it.
```

Comment:
what's "declare type" here?

Reply:
Java support polymorphism, if we declare a field with type `List<SomeClass>`, the element type may be `ChildClass`

Reply:
I see. Maybe you can change it to declared type or compile-time type.

Reply:
Updated to `declared type`